server/src/services/getAchievements.ts

```diff
@@ -0,0 +1,45 @@
+import { GetAchievements } from './../types/interfaces';
+import { allAchievements } from '../mockdata/achivements.json';
+
+const allAchievementsList = allAchievements;
+
+export const getAchievements: GetAchievements = (
+  challengeId: number,
+  challenges
+) => {
+  const currentChallenge = challenges.find(
+    (i) => i.challengeId === challengeId
+  );
+
+  if (!currentChallenge) {
+    throw new Error('required challenge does not exsit!');
+  }
+
+  const result = [];
+
+  const achievementsStatus = currentChallenge.achievementsStatus;
+
+  for (const achievement in achievementsStatus) {
```

Comment:
Please change it so:
1. You would not use any external data.
2. You would have a separate formatting function to make achievementsStatus[] into actualAchievementsStatus[]